Makefile

```diff
@@ -63,16 +84,19 @@ cover:
 	done
 
 test:
-	go test -race ./...
+	@govendor test -race +local
 
 docker_build:
 	docker run --rm -v "$$PWD":/go/src/github.com/mesosphere/etcd-mesos \
 		-e GOPATH=/go \
 		-w /go/src/github.com/mesosphere/etcd-mesos \
-		golang:1.4.2 make
+		golang:1.8.1 \
+		make
 
 docker: docker_build
-	docker build -t $(DOCKER_ORG)/etcd-mesos:$(VERSION) .
+	docker build --no-cache -t $(DOCKER_ORG)/etcd-mesos:$(VERSION) .
+	docker push $(DOCKER_ORG)/etcd-mesos:$(VERSION)
```

Comment:
i'm leery of automatically pushing. please gate this by an overridable flag, e.g:
```
	test "${DOCKER_PUSH_ENABLED}" = "" || docker push ...
```